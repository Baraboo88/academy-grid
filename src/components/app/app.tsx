import { ThemeProvider } from 'styled-components';
import {
  Switch,
  Route,
  BrowserRouter as Router,
} from '../common/common';

import DetailedQuest from '../detailed-quest/detailed-quest';
import Contacts from '../contacts/contacts';
import Home from '../home/home';
import { appTheme } from './common';
import * as S from './app.styled';
import React from 'react';
import NotFound from '../not-found/not-found';

const App = () => (
  <ThemeProvider theme={appTheme}>
    <S.GlobalStyle />
    <Router>
      <Switch>
        <Route exact path="/detailed-quest/:id" component={DetailedQuest}/>
        <Route exact path="/contacts">
          <Contacts />
        </Route>
        <Route exact path="/" >
          <Home />
        </Route>
        <Route>
          <NotFound/>
        </Route>
      </Switch>
    </Router>
  </ThemeProvider>
);

export default App;
