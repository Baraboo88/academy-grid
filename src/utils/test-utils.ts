import { QuestLevel, QuestModel, QuestType } from './utils';
import { createApi } from '../api';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';


export const mockQuests: QuestModel[]  = [
    {
      id: 1,
      "title": "Склеп",
      "description": "Средневековое кладбище таит в себе много страшных тайн. Местные жители говорят, что в склепе похоронен граф вампир, который по ночам выходит на охоту, чтобы испить человеческой крови. Через час солнце опустится за горизонт, успеете ли вы убить вампира и выбраться из склепа?",
      "previewImg": "img/preview-sklep.jpg",
      "coverImg": "img/cover-sklep.jpg",
      type: "horror" as QuestType,
      level: "hard" as QuestLevel,
      peopleCount: [2, 5],
      duration: 120
    },
    {
      "id": 2,
      "title": "Маньяк",
      "description": "В комнате с приглушённым светом несколько человек, незнакомых друг с другом, приходят в себя. Никто не помнит, что произошло прошлым вечером. Руки и ноги связаны, но одному из вас получилось освободиться. На стене висит пугающий таймер и запущен отсчёт 60 минут. Сможете ли вы разобраться в стрессовой ситуации, помочь другим, разобраться что произошло и выбраться из комнаты?",
      "previewImg": "img/preview-maniac.jpg",
      "coverImg": "img/cover-maniac.jpg",
      "type": "horror" as QuestType,
      "level": "medium" as QuestLevel,
      "peopleCount": [3, 6],
      "duration": 90
    },
    {
      "id": 3,
      "title": "Ритуал",
      "description": "Тяжелый воздух угнетает, в ночи вы оказываетесь запертыми в сыром помещении вместе с другими ничего не понимающими жертвами. Сквозь щель в двери вы видите, как некто в капюшоне готовит площадку как будто для проведения мистического обряда. Удастся ли вам выбраться, пока вы не станете жертвой ритуала?",
      "previewImg": "img/preview-ritual.jpg",
      "coverImg": "img/cover-ritual.jpg",
      "type": "mystic" as QuestType,
      "level": "hard" as QuestLevel,
      "peopleCount": [3, 5],
      "duration": 120
    },
    {
      "id": 4,
      "title": "Тайны старого особняка",
      "description": "Погрузитесь в атмосферу служебных помещений закулисья, которые хранят множество тайн и загадок. Вы окажитесь в старом особняке и увидите все, что скрывают его запутанные коридоры.",
      "previewImg": "img/preview-final-frontier.jpg",
      "coverImg": "img/cover-final-frontier.jpg",
      "type": "detective" as QuestType,
      "level": "easy" as QuestLevel,
      "peopleCount": [2, 5],
      "duration": 60
    },
    {
      "id": 5,
      "title": "Хижина в лесу",
      "description": "Вы с друзьями оказались в заброшенной хижине. Какую тайну она скрывает и как из неё выбраться? На эти вопросы вам предстоит найти ответ, чтобы вернуться домой.",
      "previewImg": "img/preview-house-in-the-woods.jpg",
      "coverImg": "img/cover-house-in-the-woods.jpg",
      "type": "mystic" as QuestType,
      "level": "medium" as QuestLevel,
      "peopleCount": [4, 7],
      "duration": 90
    },
    {
      "id": 6,
      "title": "Фатальный эксперимент",
      "description": "Вы стоите на пороге нового научного открытия, которое перевернет судьбу человечества. Но что-то идёт не так, и ядерный реактор, который работает на полную мощность, сигнализирует о скорой поломке. Удастся ли вам починить его в отведенное время и предотвратить гибель людей в этом фатальном эксперименте?",
      "previewImg": "img/preview-fatal-exp.jpg",
      "coverImg": "img/cover-fatal-exp.jpg",
      "type": "adventures" as QuestType,
      "level": "hard" as QuestLevel,
      "peopleCount": [5, 8],
      "duration": 120
    },

  ]
;


export const getTestStore = () => {
  const initialState = {
    quests: mockQuests,
    currentQuest: mockQuests[1],
    errorMsg: ''
  };
  const reducer = (state = initialState) => {
    return state;
  };
  const api = createApi();

  return createStore(reducer, applyMiddleware(thunk.withExtraArgument(api)));
};
