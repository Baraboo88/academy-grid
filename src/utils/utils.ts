import { ActiveTab } from '../reducer/reducer';

export enum QuestType {
  Adventures = 'adventures',
  Horror = 'horror',
  Mystic = 'mystic',
  Detective = 'detective',
  SciFi='sci-fi'
}

export enum QuestLevel {
  Easy = 'easy',
  Medium = 'medium',
  Hard = 'hard'
}

export enum QuestDuration {
  Sixty = 60,
  Ninety = 90,
  OneTwoZero = 120
}

export interface QuestModel {
  id: number;
  title: string;
  description: string;
  previewImg: string;
  coverImg: string;
  type: QuestType;
  level: QuestLevel;
  peopleCount: number [];
  duration: QuestDuration;
}

export interface OrderModel {
  name: string;
  peopleCount: number;
  phone: string;
  isLegal: boolean
}

export interface StateModel {
  quests: QuestModel [];
  currentQuest?: QuestModel;
  isOrderSent?: boolean;
  isResponseReceived: boolean;
  errorMsg: string;
  activeTab: ActiveTab;
}


export const getCyrillicLevel = (level: QuestLevel) => {
  switch (level) {
    case QuestLevel.Easy:
      return 'простой';
    case QuestLevel.Medium:
      return 'средний';
    case QuestLevel.Hard:
      return 'сложный';
  }
  return '';
};

export const getCyrillicType = (questType: QuestType) => {
  switch (questType) {
    case QuestType.Adventures:
      return 'Приключения';
    case QuestType.Horror:
      return 'Ужасы';
    case QuestType.Mystic:
      return 'Мистика';
    case QuestType.Detective:
      return 'Детектив';
    case QuestType.SciFi:
      return 'Sci-fi';
  }
  return '';
};


export const applyFilter = (quests: QuestModel[], filter: QuestType) => quests.filter((quest) => quest.type === filter);
