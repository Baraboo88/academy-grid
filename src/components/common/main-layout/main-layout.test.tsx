import * as Enzyme from 'enzyme';
import { mount } from 'enzyme';
import EnzymeReactAdapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';

import toJson from 'enzyme-to-json';

import { ThemeProvider } from 'styled-components';
import { appTheme } from '../../app/common';
import { getTestStore, testInitialState } from '../../../utils/test-utils';

import { BrowserRouter } from '../common';
import MainLayout from './main-layout';
import 'jest-styled-components';

Enzyme.configure({ adapter: new EnzymeReactAdapter() });

it(`MainLayout successfully rendered`, () => {
  const tree = mount(<Provider store={getTestStore()}><BrowserRouter><ThemeProvider
    theme={appTheme}><MainLayout ><div/></MainLayout></ThemeProvider></BrowserRouter></Provider>);
  expect(toJson(tree, { mode: `deep` })).toMatchSnapshot();
});
