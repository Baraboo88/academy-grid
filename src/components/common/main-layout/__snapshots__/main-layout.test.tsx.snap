// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout successfully rendered 1`] = `
Array [
  .c1 {
  max-width: 1366px;
  width: 100%;
  margin: 0 auto;
}

.c0 {
  z-index: 5;
  position: fixed;
  top: 0;
  left: 50%;
  width: 100vw;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  background-color: rgba(26,26,26,0.9);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  min-height: 74px;
  padding-top: 23px;
  padding-right: 33px;
  padding-bottom: 23px;
  padding-left: 32px;
}

.c3 {
  margin-right: 250px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  max-width: 600px;
  margin: 0;
  padding: 0;
  padding-top: 17px;
  padding-bottom: 17px;
  margin-bottom: -10px;
  list-style: none;
}

.c5:not(:last-child) {
  margin-right: 49px;
  margin-bottom: 10px;
}

.c6 {
  display: block;
  max-width: 100px;
  font-size: 14px;
  line-height: 16px;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  font-weight: 600;
  text-transform: uppercase;
  color: #F0F0F0;
  color: #F2890F;
}

.c6:focus,
.c6:hover {
  color: #F2890F;
}

.c7 {
  display: block;
  max-width: 100px;
  font-size: 14px;
  line-height: 16px;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  font-weight: 600;
  text-transform: uppercase;
  color: #F0F0F0;
}

.c7:focus,
.c7:hover {
  color: #F2890F;
}

.c8 {
  margin-top: 17px;
  margin-left: auto;
  font-size: 14px;
  line-height: 16px;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  font-weight: 600;
  color: #F0F0F0;
  font-feature-settings: 'pnum' on,'lnum' on;
}

.c8:focus,
.c8:hover {
  color: #F2890F;
}

@media (max-width:1300px) {
  .c3 {
    margin-right: auto;
  }
}

<header
    className="c0"
  >
    <div
      className="c1 c2"
    >
      <a
        className="c3"
      >
        <img
          alt="Логотип Escape Room"
          className=""
          height="50"
          src="logo.svg"
          width="134"
        />
      </a>
      <nav
        className=""
      >
        <ul
          className="c4"
        >
          <li
            className="c5"
          >
            <a
              className="c6"
              href="/"
              onClick={[Function]}
            >
              Квесты
            </a>
          </li>
          <li
            className="c5"
          >
            <a
              className="c7"
              data-test="test-header-link-click"
              href="/"
              onClick={[Function]}
            >
              Новичкам
            </a>
          </li>
          <li
            className="c5"
          >
            <a
              className="c7"
              data-test="test-header-link-click"
              href="/"
              onClick={[Function]}
            >
              Отзывы
            </a>
          </li>
          <li
            className="c5"
          >
            <a
              className="c7"
              data-test="test-header-link-click"
              href="/"
              onClick={[Function]}
            >
              Акции
            </a>
          </li>
          <li
            className="c5"
          >
            <a
              className="c7"
              href="/contacts"
              onClick={[Function]}
            >
              Контакты
            </a>
          </li>
        </ul>
      </nav>
      <a
        className="c8"
        href="tel:88003335599"
      >
        8 (800) 333-55-99
      </a>
    </div>
  </header>,
  @media (max-width:1300px) {

}

<div />,
  .c0 {
  z-index: 5;
  position: fixed;
  bottom: 24px;
  left: 32px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  max-width: 500px;
  margin: 0;
  margin-bottom: -7px;
  padding: 0;
  list-style: none;
}

.c2:not(:last-of-type) {
  margin-right: 7px;
  margin-bottom: 7px;
}

.c3 {
  display: block;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 image {
  -webkit-filter: grayscale(1);
  filter: grayscale(1);
}

.c4:focus image,
.c4:hover image {
  -webkit-filter: grayscale(0);
  filter: grayscale(0);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 circle {
  fill: #c6c4c2;
}

.c5 path {
  fill: #1A1A1A;
}

.c5:focus circle,
.c5:hover circle {
  fill: #4f779f;
}

.c5:focus path,
.c5:hover path {
  fill: white;
}

@media (max-width:1300px) {

}

<footer
    className="c0"
  >
    <ul
      className="c1"
    >
      <li
        className="c2"
      >
        <a
          className="c3"
          href="https://instagram.com"
          target="_blank"
        >
          <svg
            className="c4"
          >
            icon-instagram-hover.svg
          </svg>
          <span
            className="visually-hidden"
          >
            Инстаграм
          </span>
        </a>
      </li>
      <li
        className="c2"
      >
        <a
          className="c3"
          href="https://vk.com"
          target="_blank"
        >
          <svg
            className="c5"
          >
            icon-vk-hover.svg
          </svg>
          <span
            className="visually-hidden"
          >
            Вконтакте
          </span>
        </a>
      </li>
    </ul>
  </footer>,
]
`;
